File: src/ros2_bridge/mappers/std_msgs/float32.rs
//! Typed mapper for `std_msgs/msg/Float32`.

use crate::ros2_bridge::mapper::TypedTopicMapper;

pub(crate) fn float32_to_bus(
    msg: ros_env::std_msgs::msg::Float32,
) -> crate::std_msgs::msg::v1::Float32 {
    crate::std_msgs::msg::v1::Float32 { data: msg.data }
}

pub(crate) fn float32_to_ros(
    bus: crate::std_msgs::msg::v1::Float32,
) -> ros_env::std_msgs::msg::Float32 {
    ros_env::std_msgs::msg::Float32 { data: bus.data }
}

#[derive(Clone, Copy, Debug, Default)]
pub struct StdMsgsFloat32Mapper;

impl TypedTopicMapper for StdMsgsFloat32Mapper {
    type Ros = ros_env::std_msgs::msg::Float32;
    type Bus = crate::std_msgs::msg::v1::Float32;

    fn ros_to_bus(&self, msg: Self::Ros) -> crate::errors::Result<Self::Bus> {
        Ok(float32_to_bus(msg))
    }

    fn bus_to_ros(&self, msg: Self::Bus) -> crate::errors::Result<Self::Ros> {
        Ok(float32_to_ros(msg))
    }
}

File: bindings/python/robot_bus/ros2_bridge/mappers/std_msgs/float32.py
"""Generated mapper for `std_msgs/msg/Float32`."""

from __future__ import annotations

from robot_bus.ros2_bridge.mappers import _convert

_BusMsg = None


def _bus_cls():
    global _BusMsg
    if _BusMsg is None:
        from robot_bus.std_msgs.msg.v1 import Float32 as BusMsg

        _BusMsg = BusMsg
    return _BusMsg


def float32_to_bus(msg):
    BusMsg = _bus_cls()
    bus = BusMsg()
    bus.data = msg.data
    return bus


def float32_to_ros(bus):
    from std_msgs.msg import Float32 as RosMsg

    out = RosMsg()
    out.data = bus.data
    return out


class StdMsgsFloat32Mapper:
    _ros_type = None

    def ros_msg_type(self):
        cls = type(self)
        if cls._ros_type is None:
            from std_msgs.msg import Float32 as RosMsg

            cls._ros_type = RosMsg
        return cls._ros_type

    def ros_to_bus(self, msg) -> bytes:
        return float32_to_bus(msg).SerializeToString()

    def bus_to_ros(self, payload: bytes):
        BusMsg = _bus_cls()
        bus = BusMsg()
        bus.ParseFromString(payload)
        return float32_to_ros(bus)
